### DISCUSSION TITLE
Unable to see Tag details in Releasenotes.md file

### DISCUSSION THREAD

**Discussion by khopithan** [rfennell/AzurePipelines#1315] [Q&A] [answered]
Hi Richard,

Thank you for the immediate reply in Generate Release Notes (Crossplatform) review section. I am actually new to azure devops. I just need to print Tag details of the running pipeline. I would appreciate if you complete this code and help to print the tag value of the source code. Thanks

{{#forEach buildDetails.Tags}}
this
{{/foreach}}

**Comment by rfennell** [comment]
I just checked the usage (using my [command line tester](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole) very useful for debugging templates) and it seems my memory was wrong, you don't have to handle the build tags as an array

If I use the template

```
# {{buildDetails.definition.name}} {{buildDetails.buildNumber}}
* **Branch**: {{buildDetails.sourceBranch}}
* **Tags**: {{buildDetails.tags}}
* **Completed**: {{buildDetails.finishTime}}
```

I get the output

> # Validate - DevLabs Task 10002
> * **Branch**: refs/heads/master
> * **Tags**: Build tag,Another tag
> * **Completed**: Fri Jul 08 2022 14:27:14 GMT+0100 (British Summer Time)

If you want to reformat this list of tags you will need to use the [Handlebars Helpers](https://github.com/helpers/handlebars-helpers) I ship in the extension, or write a [custom extension](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task)

**Reply by khopithan** [reply]
Hi @rfennell,

Thank you for the reply.

Actually, I used the below code from here [https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task](url)

## Build {{buildDetails.buildNumber}}
* **Branch**: {{buildDetails.sourceBranch}}
* **Tags**: {{buildDetails.tags}}
* **Completed**: {{buildDetails.finishTime}}
* **Previous Build**: {{compareBuildDetails.buildNumber}}

But I am getting this output...so I am confused that what I am missing.

## Build devops_test.18225
* **Branch**: refs/heads/devops_test
* **Tags**: 
* **Completed**: 
* **Previous Build**: devops_test.18221

**Comment by rfennell** [comment]
All my task does is call the Azure DevOps REST API to get the build details.

For example, I have a test build with two tags

![image](https://user-images.githubusercontent.com/324718/181723596-e2c151ee-74af-453b-8354-cd5538c3e7ad.png)

The task is calling the REST API via https://dev.azure.com/richardfennell/GitHub/_apis/build/builds/10002

Which return

![image](https://user-images.githubusercontent.com/324718/181723467-70d0bded-7fff-4323-988a-c22f6c70bcaa.png)


I would suggest checking what you see with the REST call

https://dev.azure.com/<org>/<project>/_apis/build/builds/<ID>

Just to check a couple of points

- When are your build tags created, remember the task is running inside the build (this is why you have no finish time as the build is not finished). Are the release notes being generated before the build tag is create?
- My task only supports Azure DevOps WI and Build tags - there is no support for Git tags

**Reply by khopithan** [reply]
Thank you @rfennell.

I'll check this and let you know👍

**Comment by khopithan** [comment]
Hi @rfennell,

I checked the Azure API endpoint for the particular build but there is no results for tag

![image_2022-08-01_105928756](https://user-images.githubusercontent.com/23405361/182078390-f336b649-ca1b-49ab-9e59-9295394c526c.png)

And there is no workItems as well

![image_2022-08-01_110312318](https://user-images.githubusercontent.com/23405361/182078809-6bb6d84d-a45e-4562-8825-7cffedc3aaa4.png)

**Comment by rfennell** [comment]
My task can only return the results from the API, and as far as that is concerned there are no Azure DevOps Build tags associated with this build. 

As for the work items, you use the [get work items refs API call](https://docs.microsoft.com/en-us/rest/api/azure/devops/build/builds/get?view=azure-devops-rest-7.1) to get those. So I would not expect them to be present with this call.

**Comment by khopithan** [comment]
When I trigger a build by creating a new tag, it is shown by "**buildDetails.sourceBranch**" variable

![image_2022-08-01_142004694](https://user-images.githubusercontent.com/23405361/182110893-3d9708b7-5d27-4081-b574-204c2b35653c.png)

This is the releasenotes.md file output

## Build 0.0.1996.18289
* **Branch**: refs/tags/0.0.1996
* **Tags**: 
* **Completed**: 
* **Previous Build**: devops_test.18287

It didn't get any value from "**buildDetails.tags**" variable

**Comment by rfennell** [comment, marked as answer]
You are confusing Git Tags and Azure DevOps Build Tags - they are not the same thing and are unrelated.

- Git Tags are set via the Git command line tools (or your IDE) - related to source control only
- Azure DevOps Pipeline tags are set via the UI , CLI or the API - related to the pipeline only

And to add more complexity Azure DevOps WI tags are different too.

If you want the Git Tag in your release notes, I would suggest extracting it from the source branch field you highlighted either a the whole branch name, or use Handlebars functions to parse the string

**Reply by khopithan** [reply]
Thank you @rfennell for the explanation.🙂👍
